### ISSUE TITLE
Default value "" (empty string) or empty array treated as null (no default value)

### ISSUE THREAD

**Issue by Dador** [bizley/yii2-migration#49]
Hello.
Column in database has `""` (empty string) as default value, but yii2-migration treat it as null (as if there is no default value specified).

Example table (SQL):
```sql
CREATE TABLE test (
	id serial NOT NULL,
	"name" varchar(255) NOT NULL DEFAULT '',
	CONSTRAINT test_pkey PRIMARY KEY (id)
);
```

Generated code:
```php
'name' => $this->string()->notNull(),
```

Expected generated code:
```php
'name' => $this->string()->notNull()->defaultValue(''),
```

I'm using PostgreSQL. The issue itself is not database specific, but IIRC, MySQL automatically gives default value `""` (empty string) to `not null` `varchar` column. In contrast, in PostgreSQL you have to specify that you want `""` (empty string) to be default (if you don't do this PostgreSQL will raise an error if you save it without specifying value for that column).

That's important when you have `not null` column but the column itself is optional. Without default value in database, you have to specify `""` in model every time you save new model. There is a reason why column is `not null` and not `nullable`.

So, moving to the cause of issue. Changing [Generator.php:186](https://github.com/bizley/yii2-migration/blob/3.2.0/src/Generator.php#L186)
```php
'default' => $column->defaultValue ?: null,
```
to
```php
'default' => $column->defaultValue,
```
resolves the issue.
But as I see in history, it was [intentionally changed](https://github.com/bizley/yii2-migration/commit/95efa2fa48abc844b4451af20d24742a583de6df#diff-7961978656794739d73aa8a22eb6b8bcL163). Is there a reason for this?

P.S. Same thing happens with empty array for json/jsonb column. But json/jsonb also has another issue #50 .

**Comment by bizley** [comment]
IIRC it was done because of the issues with comparing empty values during update process. But definitely it needs to be looked at. Thank you for report.

**Comment by bizley** [comment]
Fixed in https://github.com/bizley/yii2-migration/commit/bc02e3fda102b964fdf4dda1f84a0e8065d20347

**Comment by bizley** [comment]
Fixed in [3.3.0](https://github.com/bizley/yii2-migration/releases/tag/3.3.0) / [2.6.0](https://github.com/bizley/yii2-migration/releases/tag/2.6.0)
